### ISSUE TITLE
Question or Issue : network or sound management 

### ISSUE THREAD

**Issue by rcmn** [lxde/lxqt#533]
I'm probably out of my depth about this. And I might have miss a conversation somewhere. So I apologize in advance if I am. 
I have been  following the project for a while now and used lxqt daily since 0.7 .
I know lately a lot of focus is put on the systray and plug ins. So that people can contribute to build those for their applications. 

My question \ issue is this : Some effort have been put towards a monitor setting manager and other component. So that make me think lxqt consider them as "core"(for lack of better words) . But I have been facing a dilemma regarding network management and sound management. For example Manjaro  suggest few options like CMST. But then it doesn't start at start up so I have to add it to the start up menu and they open in full screen when i login...not great (the plugin\systray effort will help with that I understand) . 
Is lxqt will not consider network and sound management "core" and let the different distributions pick their own ? Or will it have them "included" in a later release ?

**Comment by jhg** [comment]
I have same problem, and I am using CMST until LXQt include a own network manager, I think it's very important for "normal users" that only want use it, without problems. And with CMST I have the problem that with "autostart" when I reboot CMST say that don't find systemtray icon and that I check if systemtray is installed.

**Comment by pmattern** [comment]
You guys may want to have a look at https://github.com/surlykke/lxqt-connman-applet, available as Arch Linux AUR package [lxqt-connman-applet-git](https://aur.archlinux.org/packages/lxqt-connman-applet-git) already. (Just mentioning as I stumbled upon both this issue and that project, didn't test the latter myself yet.)

@rcmn
Regarding sound you don't state details. So what are you missing? You are aware of plugin Volume control and do know that e. g. pavucontrol can be invoked from that plugin by clicking on 'Mixer'?

**Comment by Esclapion** [comment]
I tried without success because I use the non-git versions. And if I modify the PKGBUILD to depend of liblxqt standard, the patch included doesn't work.

For cmst, the last version works fine for me with checking systemtray (why not ?). But this tray is very slow to start after the desktop, I don't know why, but it's probably another problem.

**Comment by rcmn** [comment]
@pmattern 

I'll look into the connman applet. Thanks.

Yes i'm aware of the plugin. But I don't use it anymore, because the volume manager I use have their own systray options. It seemed redundant. 

And their is this thread i completely forgot about https://github.com/lxde/lxqt/issues/369.It's a discussion regarding a QT fork of pavucontrol .At the end pranavg189 said he was working on a fork called lxmixer https://github.com/pranavg189/lxmixer .It look like the last activity was on Jan 27.

i found that kmix was working well for me for now... It has ,surprisingly, very little dependencies(only 1 , but my install is few month old so I don't know If I have already installed other dependencies thru other applications - i do not use any other KDE app however!). the good : qt base,systray,clean startup,capture applications streams. The bad: very KDE centric of course and maybe not as complete as pavucontrol.

**Comment by surlykke** [comment]
As the name indicates, my intention with lxqt-connman-applet is to propose it for inclusion in lxqt. 
The state is that it can show/disable/enable technologies (ethernet, wifi, bluetooth), show/connect/disconnect services. (Bluetooth pairing must be done by other means though (like blueman..)) What I still need from it, and am working on, is creating vpn connections. I hope to make it able to create and activate an open-vpn connection within a few weeks. I've used lxqt-connman-applet for a couple of months now, and for my (admittedly simple) use cases it seems sufficient. One nice thing about connman is that more than one client can be connected, so if you need a feature not supported by lxqt-connman-applet,  you can run connmanctl and do it from there.
@Esclapion  : Feel free to raise an issue.

**Comment by rcmn** [comment]
I would really like to use lxqt-connman-applet but I use the ubuntu PPA. Has anyone build a pkg for it by any chances?

**Comment by surlykke** [comment]
Den 31/03/2015 11.05 skrev "rcmn" <notifications@github.com>:
>
> I would really like to use lxqt-connman-applet but I use the ubuntu PPA.
Has anyone build a pkg for it by any chances?
>
> —

https://aur.archlinux.org/packages/lxqt-connman-applet-git/

**Comment by rcmn** [comment]
@surlykke  Yep I've seen that. @pmattern mentionned the AUR package few days ago. But since you are reposting it I must be missing something !?

**Comment by surlykke** [comment]
Den 31/03/2015 21.43 skrev "rcmn" <notifications@github.com>:
>
> @surlykke Yep I've seen that. @pmattern mentionned the AUR package few
days ago. But since you are reposting it I must be missing something !?

No no. Hadn't seen that pmattern mentioned it. Can't you use that one?

Br. Chr.

**Comment by rcmn** [comment]
Well I'm really not familliar with AUR pkg. But I'm not afraid to learn so
I search for ways to either use directly the AUR pkg as is on my debian
based system(since it seems to be a simple script) or find a tool that
would convert it easily into a compatible one(like makepkg). But neither
solution seemed very straight forward and they seemed like  hazardous
processes with a slim chances of an happy outcome. Installing pacman
alongside dpkg doesn't seems worth it either.
So maybe there is a simple obvious way that I'm not seeing.

**Comment by pmattern** [comment]
Just saw that parts of this issue have already been addressed in #125 (network management GUI in general) as well as #297 (Bluetooth GUI).

**Comment by agaida** [comment]
Will make a debian package for it this week

**Comment by rcmn** [comment]
@agaida

**Comment by agaida** [comment]
@surlykke - i think its time to move/fork lxqt-connman-applet to g.c/lxde - what do you think?

**Comment by surlykke** [comment]
2015-04-22 1:10 GMT+02:00 Alf Gaida <notifications@github.com>:

> @surlykke <https://github.com/surlykke> - i think its time to move/fork
> lxqt-connman-applet to g.c/lxde - what do you think?
>
I think so too.  I was hoping to do (some of the) vpn stuff before asking
to move it into lxqt. But realistically it will be several weeks before I
get around to it. So now may be as good a time as any.

br. Chr

**Comment by agaida** [comment]
ok, experimental packaging is done (vs Qt 5.4.1), packaging for unstable will follow soon (today). Work well, shows a Symbol in the tray with a working menu. The only thing i would change is: Please make the applet quiet, aka no debug output in standard. Maybe a command ĺine would be nice. A version would be nice too.
One could find the packages here: http://packages.siduction.org/lxqt - the experimental packages need parts from experimental, for the unstable packages the repo should be sufficient.

**Comment by rcmn** [comment]
@agaida just installed and tried this package "lxqt-connman-applet_0.0.0-1_amd64.deb 22-Apr-2015 19:51 88668" on Ubuntu 14.10. 

@surlykke Tested wired and wifi (with adding passphrase) everything worked. 
The only thing I had to do is to add "lxqt-connman-applet" to autostart.
One suggestion / detail for the futur, something that cmst does not have either, is some sort of visual indicator when connman is working in the background, with a message of failure if needed. Example:  my wifi configuration took more time that I was expecting and in both application (cmst and lxqt-connman-applet) I was turning off wifi and on again too early trying to troubleshoot. It's only after launching the application from the console I knew I had to wait. 

thank you!

**Comment by Vladimir-csp** [comment]
Any plans on Network Manager support?

**Comment by palinek** [comment]
@paulolieuthier is your plugin (noticed in #125) an NM frontend?

**Comment by paulolieuthier** [comment]
@palinek yes, and it uses [NetworkManager-qt](http://api.kde.org/frameworks-api/frameworks5-apidocs/networkmanager-qt/html/index.html). However, it's still going to take a lot of time for me to call it usable.

**Comment by palinek** [comment]
Should anyone be interested.... 
I've been working on a simple [app](https://github.com/palinek/nm-tray) on top of NetworkManager-qt. Currently it is everything but not finished yet. But already allows connecting to unsecured or WpaPsk wifi and viewing some information.

Any thoughts, reviews, help are welcome.

**Comment by MttCastelli** [comment]
I'm not sure that this matter, but on my box (an up to date lubuntu xenial with lubuntu-dev ppa, total lxqt desktop with connman to repace nm) I have to shout down and than up wireless (trough lxqt-connman-applet) at every boot to get it works. Does anybody have or have had the same issue?

The nearest state I've found for this is in it's docs https://01.org/connman/documentation: 

> ConnMan will remember the state of the network technologies over reboots. By default all technologies are disabled at the very first startup in order to prevent unwanted wireless or wired communication from happening. Technologies can be enabled and disabled with:
> 
> > connmanctl enable wifi
> Enabled wifi
> 
> > connmanctl disable wifi
> Disabled wifi

..but here is told about the very first boot, while at every boot I have wireless down.

**Comment by agaida** [comment]
one could use cmst as a connman frontend - and one should use the latest available connman. Right now it is 1.32 afaik.

**Comment by pmattern** [comment]
@MttCastelli 
If you don't fear the command line you may consider using `connmanctl` to narrow this problem down as well, eventually after removing LXQt ConnMan applet. If invoked without any arguments there's a genuine shell, the command `help` of that shell should get you started.

Note the problem basically isn't related to this issue. So if further discussion is needed it should probably better take place elsewhere in order to avoid cluttering this issue.

**Comment by MarkG-108** [comment]
Regarding network management, I'm using Debian Bookworm (Debian Linux 12).  I have lxqt installed, and it is the sole window and/or desktop manager on my system.  Internet connectivity relies upon wpa_supplicant.  I have connman installed, which, via cmst, provides an applet in the panel.  However, this applet does not reflect what is going on in my system.  It will tell me that the state is offline, even when I am connected.  

When I first installed my system, I was able to get WiFi working via connman, and thus was able to set up the connection.  But, sometimes it boots without establishing a WiFi connection, and in these instances I am unable to re-establish the connection with connman/cmst.  The only measure I can take in such a circumstance is to reboot and hope a WiFi connection is established.

I tried installing wpa_gui, which likewise provides a panel applet.  This seems to better reflect what is going on, but it too will not re-establish a WiFi connection that was not properly established at boot time.  Thus, a reboot is required.

Is there a way to have it so that I can establish a WiFi connection without rebooting?

**Comment by stefonarch** [comment]
> It will tell me that the state is offline, even when I am connected.

That is  a known bug in connman, restarting it it will display state correctly.

>Is there a way to have it so that I can establish a WiFi connection without rebooting?

closing cmst and `# systemctl restart connman.service`  and reoping cmst?

**Comment by MarkG-108** [comment]
I did find a way, that being by running the command `dpkg-reconfigure wpasupplicant`.

The internet connection is quite erratic though.
